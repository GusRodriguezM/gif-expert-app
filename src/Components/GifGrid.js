import React, {useEffect} from 'react';

const GifGrid = ({category}) => {

    useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async() => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=xzHdGrQXHRj14LP2GM7YF0eUh3z1Rcwq';
        const resp = await fetch(url);
        const {data} = await resp.json();
        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });
    }

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}

export default GifGrid;