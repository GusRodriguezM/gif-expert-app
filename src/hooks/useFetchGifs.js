import {getGifs} from '../helpers/getGifs';
import {useEffect, useState} from 'react';

export const useFetchGifs = (category) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        getGifs(category)
            .then(imgs => {

                setTimeout(() => {
                    
                    setState({
                        data: imgs,
                        loading: false
                    });

                }, 3000);

        });

    }, [category]);

    return state;

}
