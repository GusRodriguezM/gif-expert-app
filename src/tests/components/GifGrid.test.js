import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests in <GifGrid/> component', () => {

    const category = 'Dragon Ball';
   
    test('Should render <GrifGrid/> correctly', () => {
       
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

    });
    
    test('Should render items when images are loaded with useFetchGifs custom hook ', () => {

        const gifs = [{
            id: 'ABC123',
            url: 'https://localhost/any/gif',
            title: 'Any Gif'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

    });
    
    
});
