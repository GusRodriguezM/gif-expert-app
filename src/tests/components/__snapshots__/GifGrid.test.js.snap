// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid/> component Should render <GrifGrid/> correctly 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__pulse"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests in <GifGrid/> component Should render items when images are loaded with useFetchGifs custom hook  1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="Any Gif"
      url="https://localhost/any/gif"
    />
  </div>
</Fragment>
`;
